app/src/main/java/app/grapheneos/camera/capturer/ImageCapturer.kt

```diff
@@ -29,26 +29,32 @@ var isTakingPicture: Boolean = false
 class ImageCapturer(val mActivity: MainActivity) {
     val camConfig = mActivity.camConfig
 
+    var cFadeAnim : AlphaAnimation? = null
+
     private fun fadeCaptureButton() {
         mActivity.captureButton.isEnabled = false
 
-        val animation: Animation = AlphaAnimation(mActivity.captureButton.alpha, 0.6f)
-        animation.duration = 200
-        animation.interpolator = LinearInterpolator()
-        animation.fillAfter = true
+        cFadeAnim?.cancel()
+
+        cFadeAnim = AlphaAnimation(mActivity.captureButton.alpha, 0.6f)
+        cFadeAnim?.duration = 200
+        cFadeAnim?.interpolator = LinearInterpolator()
+        cFadeAnim?.fillAfter = true
 
-        mActivity.captureButton.startAnimation(animation)
+        mActivity.captureButton.startAnimation(cFadeAnim)
     }
 
     private fun unfadeCaptureButton() {
         mActivity.captureButton.isEnabled = true
 
-        val animation: Animation = AlphaAnimation(mActivity.captureButton.alpha, 1f)
-        animation.duration = 200
-        animation.interpolator = LinearInterpolator()
-        animation.fillAfter = true
+        cFadeAnim?.cancel()
+
+        cFadeAnim = AlphaAnimation(mActivity.captureButton.alpha, 1f)
+        cFadeAnim?.duration = 200
```

Comment:
There's no need for null-safe `?.` here, it's guaranteed that `AlphaAnimation` is non-null.
Put it in a local `val`, or better yet, rewrite it like this:

```
        val button = mActivity.captureButton
        button.isEnabled = true

        cFadeAnim?.cancel()

        AlphaAnimation(button.alpha, 1f).apply {
            duration = 200
            interpolator = LinearInterpolator()
            fillAfter = true

            button.startAnimation(this)
            cFadeAnim = this
        }
```

Same applies to `fadeCaptureButton()`.

Reply:
> The cFadeAnim is expected to be null when the first fade/unfade animation gets performed. We perform the cancel operation to avoid having multiple animations run on the same view while taking consecutive shots.

I understand that.
I meant that `cFadeAnim` is guaranteed not to be null at that line (it was just assigned) . Kotlin compiler doesn't know that `cFadeAnim` is not being modified by other threads, and thus requires null checks on each access.

An alternative would be the following:
```
val anim = AlphaAnimation(button.alpha, 1f)
anim.duration = 200
...
cFadeAnim = anim
```

Reply:
> This comment was attached to line 53 (cFadeAnim?.duration = 200), not to line 50 (cFadeAnim?.cancel()).

Oh that's right, multiple null-checks could have been avoided there.

Have just pushed the required changes with the help of the `apply` method/approach.